### DISCUSSION TITLE
Show user more low-level errors

### DISCUSSION THREAD

**Discussion by offsetcyan** [magit/forge#612] [Ideas]
In my `~/.ssh/config` I have two hosts:
```
Host github github.com
  Hostname github.com
  IdentitiesOnly yes
  User git
  IdentityFile /home/ofc/.ssh/id_dch_ed25519

Host organisation-gh
  Hostname github.com
  IdentitiesOnly yes
  User git
  IdentityFile /home/ofc/.ssh/id_campari_ff
```

I have created a PAT with scopes `repo, user, read:org`.
I've added two aliases to forge-alist:
```
(setq forge-alist
        (append forge-alist
                '(("organisation-gh" "api.github.com" "github.com" forge-github-repository))
                '(("github" "api.github.com" "github.com" forge-github-repository)))))
```
When trying to M-x `forge-add-repository` I was told the host wasn't in `magit-clone-name-alist`, so I've added it there too:
```
(setq magit-clone-name-alist
      (append magit-clone-name-alist
              '(("\\`\\(?:organisation-gh:\\)?\\([^:]+\\)\\'" "organisation-gh" "github.user"))))
```

It appears `magit-forge` works in the case of:
Any repository `github:owner/repo` and a private repository my user is the owner of `organisation-gh:owner/repo`.

It does not work in the case that the private repository is owned by the organisation my user is a part of, because the repository supposedly doesn't exist:
```
Repository "<owner>/<repo>" does not exist on "api.github.com".
Maybe it was renamed and you have to update "remote.<remote>.url"?
```

**Comment by tarsius** [comment]
There's some information concerning this in https://github.com/magit/forge/wiki/Tips-and-Tricks. Also look at https://github.com/magit/forge/discussions/436.

**Reply by offsetcyan** [reply]
None of this would've helped my issue, after looking into it again today and debugging with the Emacs debugger I found this string being returned by ghub: "`<Organisation>` forbids access via a personal access token (classic). Please use a GitHub App, OAuth App, or a personal access token with fine-grained permissions."

It would be nice if this string could be raised up to the user in some clear way.

**Comment by tarsius** [comment]
We try to signal errors in `ghub--handle-response-error`, using `ghub--signal-error` and called by `ghub--handle-response`.

If you look at `ghub--read-json-payload` you will see that Github "randomly" returns html instead of json for certain errors, which makes detection difficult. I suspect this particular error falls into that category.

I don't have access to an organization that forbits personal tokens, but if you post the full response html body (preferably also the header), then I could maybe find a way to detect and forward such errors.

----

On the other hand, to be able to focus on other projects, I have to reduce the time I spend on Forge. That involves deciding to not implement certain features and I'm afraid I would rather not implement this.
